### PR TITLE
NAS-132728 / 25.04 / Add `cloud_backup.abort` test

### PR DESCRIPTION
**Pull request by creatorcary**
This is the only cloud_backup endpoint not covered by our api tests yet.

http://jenkins.eng.ixsystems.net:8080/job/tests/job/api_tests/2331/

### CODE REVIEW

File: tests/api2/test_cloud_backup.py [hunk 1]
```diff
@@ -150,7 +150,9 @@ def test_cloud_backup_abort(cloud_backup_task):
     task_id = cloud_backup_task.task["id"]
     call("cloud_backup.sync", task_id, {"dry_run": True})
     assert call("cloud_backup.abort", task_id)
+
     # Ensure backup works after an abort
+    ssh(f"touch /mnt/{cloud_backup_task.local_dataset}/testfile")
```
Comment (mgrimesix) [inline tests/api2/test_cloud_backup.py:155]: Do we validate that `testfile` was successfully backed up?

**Review by mgrimesix — APPROVED** [review]
Approve with comment.

**Review by themylogin — CHANGES_REQUESTED** [review]

File: tests/api2/test_cloud_backup.py [hunk 2]
```diff
@@ -146,6 +146,17 @@ def test_cloud_backup(cloud_backup_task):
     assert all(snapshot["id"] != snapshot_to_delete_id for snapshot in snapshots)
 
 
+def test_cloud_backup_abort(cloud_backup_task):
+    task_id = cloud_backup_task.task["id"]
+    call("cloud_backup.sync", task_id, {"dry_run": True})
```
[the remark this answers is not in the document]
Comment (creatorcary) [reply tests/api2/test_cloud_backup.py:151]: We don't actually need to backup an empty file. We could probably set `dry_run=True` for most of the tests in this module without consequence.

[the remark this answers is not in the document]
Comment (themylogin) [reply tests/api2/test_cloud_backup.py:151]: If we don't perform any writes to the backup backend, what are we testing here then?

[the remark this answers is not in the document]
Comment (creatorcary) [reply tests/api2/test_cloud_backup.py:151]: Hmm I guess we do need to make sure it's creating a lock on the repository in this case

**Comment by themylogin** [comment]
@creatorcary this still does not test anything. The cloud backup task is aborted shortly after its started, `restic` might not have even started yet. We need to actually ensure we've started backing up something huge. Having a transfer speed limiter just for testing will be a huge help to make this process more deterministic.

File: tests/api2/test_cloud_backup.py [hunk 3]
```diff
@@ -146,6 +146,27 @@ def test_cloud_backup(cloud_backup_task):
...
+    task_id = cloud_backup_task.task["id"]
+    testfile = f"/mnt/{cloud_backup_task.local_dataset}/testfile"
+
+    # Start to backup a 1G file
+    ssh(f"dd if=/dev/urandom of={testfile} bs=32M count=32")
+    job_id = call("cloud_backup.sync", task_id)
+
+    # Wait for 50% backup completion
+    while call("core.get_jobs", [["id", "=", job_id]], {"get": True})["progress"]["percent"] < 50:
+        time.sleep(0.5)
```
Comment (themylogin) [inline tests/api2/test_cloud_backup.py:160]: I would say even 0.1

**Review by themylogin — APPROVED** [review]
Thank you!

**Pull request merged**